### ISSUE TITLE
:bug: fix(bal-step-item): Add a disabled attribute to step items buttons

### ISSUE THREAD

**Issue by JulienSBaloise** [baloise/design-system#1946]
### Current Behavior

In the bal-steps component, when the "disabled" attribute of a bal-step-item is set to "true", the item is indeed visually disabled and has the "aria-disabled" attribute (Assigned to a button with "tab" role). However, this attribute alone is insufficient to prevent these elements from being focused during keyboard browsing; the "disabled" attribute must also be present.
This raises an accessibility issue because keyboard users end up focusing on disabled elements without any visual cues.

<img width="718" height="106" alt="Image" src="https://github.com/user-attachments/assets/31108a94-c3dc-4b4c-bac8-443a31ae3764" />

### Expected Behavior

bal-step-item elements should not be focusable when disabled (or focusability should be conditional via another custom attribute).

### Steps to Reproduce

Create a bal-steps element with multiple bal-step items, at least one of which is disabled.
Use keyboard browsing (Tab key).
The disabled item is now focused (unwanted).

### Version

19.8.4

### Browsers

Chrome

### Device

Desktop

### Code Reproduction URL

https://homedigital.dev-wap-lu.balgroupit.com/

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct
